pkg/pebinaryexecutor/pebinary.go

```diff
@@ -94,8 +95,19 @@ func (p *PEBinaryConfig) Bootstrap(ctx context.Context, nodeConfig *daemonconfig
 
 	restConfig, err := clientcmd.BuildConfigFromFlags("", nodeConfig.AgentConfig.KubeConfigK3sController)
 
-	if err := p.cni.Setup(ctx, nodeConfig, restConfig, p.DataDir); err != nil {
-		return err
+	switch p.CNI {
+	case "":
+		if err := p.CNIConfig.Setup(ctx, nodeConfig, restConfig, p.DataDir); err != nil {
+			return err
+		}
+	case "calico":
```

Comment:
All of the "calico", "none", etc strings should generally be defined as constants.